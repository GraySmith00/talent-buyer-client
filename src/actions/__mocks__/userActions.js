import { mockUser } from '../../Utils/mockData';

export const registerUser = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockUser);
});
