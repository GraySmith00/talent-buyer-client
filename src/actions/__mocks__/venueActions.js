import { mockVenue } from '../../Utils/mockData';

export const setNewVenue = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockVenue);
});

export const setUserVenue = jest.fn().mockImplementation(() => {
  return Promise.resolve(mockVenue);
});
