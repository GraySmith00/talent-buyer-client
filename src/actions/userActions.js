import { userSignUp } from '../Utils/backendApiCalls';

// setCurrentUserState action
export const setCurrentUserState = user => ({
  type: 'SET_CURRENT_USER_STATE',
  user
});

// registerUser thunk
export const registerUser = user => async dispatch => {
  const savedUser = await userSignUp(user);
  dispatch(setCurrentUserState(savedUser));
  return savedUser;
};
