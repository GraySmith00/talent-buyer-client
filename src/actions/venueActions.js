import { getMatchingVenue } from '../Utils/songKickApiCalls';
import { venuePostRequest, getAllUserVunues } from '../Utils/backendApiCalls';

// setCurrentVenueState action
export const setCurrentVenueState = venue => ({
  type: 'SET_CURRENT_VENUE_STATE',
  venue
});

// setNewVenue thunk
export const setNewVenue = (venueName, venueCity) => async dispatch => {
  // get venue info from songkick
  const venueInfo = await getMatchingVenue(venueName, venueCity);

  if (!venueInfo) {
    throw new Error({
      message: 'Sorry, no venue with that name was found in that city'
    });
  }

  // venue post request
  const savedVenue = await venuePostRequest(venueInfo);
  dispatch(setCurrentVenueState(savedVenue));
  return savedVenue;
};

// setUserVenue thunk
export const setUserVenue = userId => async dispatch => {
  const userVenues = await getAllUserVunues(userId);
  if (!userVenues.length) {
    throw new Error({
      message: 'Sorry, this user is not associated with any venues'
    });
  }

  dispatch(setCurrentVenueState(userVenues[0]));
  return userVenues[0];
};
