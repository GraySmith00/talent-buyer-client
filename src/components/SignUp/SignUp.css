.sign-up {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 99;
  background-color: rgba(0, 0, 0, 0.5); }
  .sign-up .header-text {
    font-family: "Open Sans", sans-serif; }
  .sign-up .inner-modal {
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 43rem;
    width: 30rem;
    background-color: #21222b;
    border-radius: 10px; }
  .sign-up .top-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 30%;
    width: 100%;
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("./concertBackground2.jpg");
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    -webkit-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75); }
  .sign-up .top-inner {
    width: 80%;
    color: #fff;
    font-size: 2.5rem; }
  .sign-up .top-inner h3 {
    margin: 0; }
  .sign-up p {
    margin: 0.8rem 0 0 0;
    color: #fff; }
  .sign-up .sign-up-form {
    margin-top: 0.8rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%; }
  .sign-up .close-text {
    color: #de0048;
    cursor: pointer;
    padding-right: 1.75em;
    padding-top: 1.75em; }
  .sign-up .close-wrap {
    display: flex;
    width: 100%;
    justify-content: flex-end; }
