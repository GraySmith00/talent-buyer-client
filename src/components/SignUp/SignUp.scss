.sign-up {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 99;

  background-color: rgba(0, 0, 0, 0.5);

  .inner-modal {
    display: flex;
    flex-direction: column;
    align-items: center;

    height: 40rem;
    width: 30rem;
    background-color: #21222b;
    border-radius: 10px;
  }

  .top-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    height: 30%;
    width: 100%;
    background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
      url('./concertBackground2.jpg');
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center center;

    border-top-left-radius: 10px;
    border-top-right-radius: 10px;

    -webkit-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    -moz-box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
    box-shadow: 0px 10px 67px -10px rgba(0, 0, 0, 0.75);
  }

  .top-inner {
    width: 80%;
    color: #fff;
    font-size: 2.5rem;
  }

  .top-inner h3 {
    margin: 0;
  }

  p {
    margin: 0.8rem 0 0 0;
    color: #fff;
  }

  .sign-up-form {
    margin-top: 0.8rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
  }

  input {
    width: 80%;
    outline: none;
    border: none;
    margin-top: 10px;
    padding: 5px 10px;
    font-size: 1.2rem;
    background-color: rgba(250, 250, 250, 0.3);
    color: #fff;
    border-radius: 5px;
  }

  button {
    margin-top: 15px;
    width: 80%;
    padding: 15px;
    border: none;
    border-radius: 5px;
    font-size: 1.2rem;
    background-color: #de0048;
    color: #fff;
    cursor: pointer;
  }

  .close-text {
    color: #de0048;
    cursor: pointer;
  }
}
