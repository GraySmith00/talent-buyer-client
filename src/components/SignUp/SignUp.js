import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { registerUser } from '../../thunks/registerUser';
import { setCurrentVenue } from '../../thunks/setCurrentVenue';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import './SignUp.css';

class SignUp extends Component {
  state = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    venueName: '',
    venueCity: ''
  };

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  handleSubmit = async e => {
    e.preventDefault();
    const savedUser = await this.saveUser();
    if (savedUser) {
      this.props.history.push('/home');
    }
  };

  saveUser = async () => {
    const { firstName, lastName, email, password } = this.state;
    const user = {
      buyer: {
        first_name: firstName,
        last_name: lastName,
        email,
        password
      }
    };
    return await this.props.registerUser(user);
  };

  render() {
    const { closeSignUpModal } = this.props;
    return (
      <form className="sign-up" onSubmit={this.handleSubmit}>
        <div className="inner-modal">
          <h1>Sign Up</h1>
          <div className="input-wrapper">
            <h4>First Name:</h4>
            <input
              type="text"
              name="firstName"
              value={this.state.firstName}
              placeholder="first name"
              onChange={this.handleChange}
            />
          </div>
          <div className="input-wrapper">
            <h4>Last Name:</h4>
            <input
              type="text"
              name="lastName"
              value={this.state.lastName}
              placeholder="last name"
              onChange={this.handleChange}
            />
          </div>
          <div className="input-wrapper">
            <h4>Email:</h4>
            <input
              type="email"
              name="email"
              value={this.state.email}
              placeholder="email"
              onChange={this.handleChange}
            />
          </div>
          <div className="input-wrapper">
            <h4>Password:</h4>
            <input
              type="password"
              name="password"
              value={this.state.password}
              placeholder="password"
              onChange={this.handleChange}
            />
          </div>
          <div className="input-wrapper">
            <h4>Venue Name:</h4>
            <input
              type="text"
              name="venueName"
              value={this.state.venueName}
              placeholder="Venue Name"
              onChange={this.handleChange}
            />
          </div>
          <div className="input-wrapper">
            <h4>Venue City:</h4>
            <input
              type="text"
              name="venueCity"
              value={this.state.venueCity}
              placeholder="Venue City"
              onChange={this.handleChange}
            />
          </div>
          <button>SUBMIT</button>
          <button>Log In</button>
          <p onClick={closeSignUpModal}>x</p>
        </div>
      </form>
    );
  }
}

SignUp.propTypes = {
  closeSignUpModal: PropTypes.func.isRequired
};

export const mapDispatchToProps = dispatch => ({
  registerUser: user => dispatch(registerUser(user)),
  setCurrentVenue: (venueName, venueCity) => dispatch(setCurrentVenue(venueName, venueCity))
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(SignUp)
);
