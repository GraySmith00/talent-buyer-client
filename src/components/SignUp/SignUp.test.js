import React from 'react';
import { shallow } from 'enzyme';
import { mockHistory } from '../../Utils/mockData';
import { setNewVenue } from '../../actions/venueActions';
import { registerUser } from '../../actions/userActions';

import { SignUp } from './SignUp';

jest.mock('../../actions/venueActions');
jest.mock('../../actions/userActions');


describe('SignUp component', () => {
  let wrapper;
  let mockCloseSignUpModal;
  let mockRegisterUser;

  beforeEach(() => {
    mockCloseSignUpModal = jest.fn();
    mockRegisterUser = jest.fn();

    wrapper = shallow(
      <SignUp
        closeSignUpModal={mockCloseSignUpModal}
        history={mockHistory}
        registerUser={mockRegisterUser}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an initial state', () => {
    const expected = {
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      venueName: '',
      venueCity: '',
      venueError: '',
      userError: ''
    };

    expect(wrapper.state()).toEqual(expected);
  });

  describe('handleChange', () => {
    it.skip('should change state when an input is changed', () => {
      wrapper.find('input').simulate('change', { target: { value: 'Bob' } });
    });
  });

  describe('saveVenue', () => {
    it.skip('should call setNewVenue', () => {
      wrapper.instance().saveVenue();
      expect(setNewVenue).toHaveBeenCalled();
    });
  });

  describe('saveUser', () => {
    it.skip('should call registerUser', () => {
      wrapper.instance().saveUser();
      expect(registerUser).toHaveBeenCalled();
    });
  });
});
