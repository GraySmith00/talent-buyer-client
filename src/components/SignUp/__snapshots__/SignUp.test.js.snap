// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    closeSignUpModal={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="inner-modal"
>
        <h1>
                Sign Up
        </h1>
        <div
                className="input-wrapper"
        >
                <h4>
                        First Name:
                </h4>
                <input
                        name="firstName"
                        onChange={[Function]}
                        placeholder="first name"
                        type="text"
                        value=""
                />
        </div>
        <div
                className="input-wrapper"
        >
                <h4>
                        Last Name:
                </h4>
                <input
                        name="lastName"
                        onChange={[Function]}
                        placeholder="last name"
                        type="text"
                        value=""
                />
        </div>
        <div
                className="input-wrapper"
        >
                <h4>
                        Email:
                </h4>
                <input
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
                />
        </div>
        <div
                className="input-wrapper"
        >
                <h4>
                        Password:
                </h4>
                <input
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
                />
        </div>
        <div
                className="input-wrapper"
        >
                <h4>
                        Venue Name:
                </h4>
                <input
                        name="venueName"
                        onChange={[Function]}
                        placeholder="Venue Name"
                        type="text"
                        value=""
                />
        </div>
        <div
                className="input-wrapper"
        >
                <h4>
                        Venue City:
                </h4>
                <input
                        name="venueCity"
                        onChange={[Function]}
                        placeholder="Venue City"
                        type="text"
                        value=""
                />
        </div>
        <button>
                SUBMIT
        </button>
        <button>
                Log In
        </button>
        <p
                onClick={[Function]}
        >
                x
        </p>
</div>,
      "className": "sign-up",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Sign Up
</h1>,
          <div
            className="input-wrapper"
>
            <h4>
                        First Name:
            </h4>
            <input
                        name="firstName"
                        onChange={[Function]}
                        placeholder="first name"
                        type="text"
                        value=""
            />
</div>,
          <div
            className="input-wrapper"
>
            <h4>
                        Last Name:
            </h4>
            <input
                        name="lastName"
                        onChange={[Function]}
                        placeholder="last name"
                        type="text"
                        value=""
            />
</div>,
          <div
            className="input-wrapper"
>
            <h4>
                        Email:
            </h4>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
</div>,
          <div
            className="input-wrapper"
>
            <h4>
                        Password:
            </h4>
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
</div>,
          <div
            className="input-wrapper"
>
            <h4>
                        Venue Name:
            </h4>
            <input
                        name="venueName"
                        onChange={[Function]}
                        placeholder="Venue Name"
                        type="text"
                        value=""
            />
</div>,
          <div
            className="input-wrapper"
>
            <h4>
                        Venue City:
            </h4>
            <input
                        name="venueCity"
                        onChange={[Function]}
                        placeholder="Venue City"
                        type="text"
                        value=""
            />
</div>,
          <button>
            SUBMIT
</button>,
          <button>
            Log In
</button>,
          <p
            onClick={[Function]}
>
            x
</p>,
        ],
        "className": "inner-modal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                First Name:
</h4>,
              <input
                name="firstName"
                onChange={[Function]}
                placeholder="first name"
                type="text"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name:",
              },
              "ref": null,
              "rendered": "First Name:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "firstName",
                "onChange": [Function],
                "placeholder": "first name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Last Name:
</h4>,
              <input
                name="lastName"
                onChange={[Function]}
                placeholder="last name"
                type="text"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Name:",
              },
              "ref": null,
              "rendered": "Last Name:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "lastName",
                "onChange": [Function],
                "placeholder": "last name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Email:
</h4>,
              <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email:",
              },
              "ref": null,
              "rendered": "Email:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Password:
</h4>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Password:",
              },
              "ref": null,
              "rendered": "Password:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Venue Name:
</h4>,
              <input
                name="venueName"
                onChange={[Function]}
                placeholder="Venue Name"
                type="text"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Venue Name:",
              },
              "ref": null,
              "rendered": "Venue Name:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "venueName",
                "onChange": [Function],
                "placeholder": "Venue Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Venue City:
</h4>,
              <input
                name="venueCity"
                onChange={[Function]}
                placeholder="Venue City"
                type="text"
                value=""
/>,
            ],
            "className": "input-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Venue City:",
              },
              "ref": null,
              "rendered": "Venue City:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "venueCity",
                "onChange": [Function],
                "placeholder": "Venue City",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SUBMIT",
          },
          "ref": null,
          "rendered": "SUBMIT",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log In",
          },
          "ref": null,
          "rendered": "Log In",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="inner-modal"
>
          <h1>
                    Sign Up
          </h1>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              First Name:
                    </h4>
                    <input
                              name="firstName"
                              onChange={[Function]}
                              placeholder="first name"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              Last Name:
                    </h4>
                    <input
                              name="lastName"
                              onChange={[Function]}
                              placeholder="last name"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              Email:
                    </h4>
                    <input
                              name="email"
                              onChange={[Function]}
                              placeholder="email"
                              type="email"
                              value=""
                    />
          </div>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              Password:
                    </h4>
                    <input
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="password"
                              value=""
                    />
          </div>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              Venue Name:
                    </h4>
                    <input
                              name="venueName"
                              onChange={[Function]}
                              placeholder="Venue Name"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="input-wrapper"
          >
                    <h4>
                              Venue City:
                    </h4>
                    <input
                              name="venueCity"
                              onChange={[Function]}
                              placeholder="Venue City"
                              type="text"
                              value=""
                    />
          </div>
          <button>
                    SUBMIT
          </button>
          <button>
                    Log In
          </button>
          <p
                    onClick={[Function]}
          >
                    x
          </p>
</div>,
        "className": "sign-up",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Sign Up
</h1>,
            <div
              className="input-wrapper"
>
              <h4>
                            First Name:
              </h4>
              <input
                            name="firstName"
                            onChange={[Function]}
                            placeholder="first name"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="input-wrapper"
>
              <h4>
                            Last Name:
              </h4>
              <input
                            name="lastName"
                            onChange={[Function]}
                            placeholder="last name"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="input-wrapper"
>
              <h4>
                            Email:
              </h4>
              <input
                            name="email"
                            onChange={[Function]}
                            placeholder="email"
                            type="email"
                            value=""
              />
</div>,
            <div
              className="input-wrapper"
>
              <h4>
                            Password:
              </h4>
              <input
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="password"
                            value=""
              />
</div>,
            <div
              className="input-wrapper"
>
              <h4>
                            Venue Name:
              </h4>
              <input
                            name="venueName"
                            onChange={[Function]}
                            placeholder="Venue Name"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="input-wrapper"
>
              <h4>
                            Venue City:
              </h4>
              <input
                            name="venueCity"
                            onChange={[Function]}
                            placeholder="Venue City"
                            type="text"
                            value=""
              />
</div>,
            <button>
              SUBMIT
</button>,
            <button>
              Log In
</button>,
            <p
              onClick={[Function]}
>
              x
</p>,
          ],
          "className": "inner-modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  First Name:
</h4>,
                <input
                  name="firstName"
                  onChange={[Function]}
                  placeholder="first name"
                  type="text"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "First Name:",
                },
                "ref": null,
                "rendered": "First Name:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "firstName",
                  "onChange": [Function],
                  "placeholder": "first name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Last Name:
</h4>,
                <input
                  name="lastName"
                  onChange={[Function]}
                  placeholder="last name"
                  type="text"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Last Name:",
                },
                "ref": null,
                "rendered": "Last Name:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "lastName",
                  "onChange": [Function],
                  "placeholder": "last name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Email:
</h4>,
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="email"
                  type="email"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email:",
                },
                "ref": null,
                "rendered": "Email:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "email",
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Password:
</h4>,
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="password"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password:",
                },
                "ref": null,
                "rendered": "Password:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Venue Name:
</h4>,
                <input
                  name="venueName"
                  onChange={[Function]}
                  placeholder="Venue Name"
                  type="text"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Venue Name:",
                },
                "ref": null,
                "rendered": "Venue Name:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "venueName",
                  "onChange": [Function],
                  "placeholder": "Venue Name",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Venue City:
</h4>,
                <input
                  name="venueCity"
                  onChange={[Function]}
                  placeholder="Venue City"
                  type="text"
                  value=""
/>,
              ],
              "className": "input-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Venue City:",
                },
                "ref": null,
                "rendered": "Venue City:",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "venueCity",
                  "onChange": [Function],
                  "placeholder": "Venue City",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "SUBMIT",
            },
            "ref": null,
            "rendered": "SUBMIT",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "x",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
